### PR TITLE
Add option to skip the unit tests (test_step)

### PR DESCRIPTION
**Pull request by Flamefire**
Add --skip-test-step to skip the test_step

### CODE REVIEW

File: test/framework/easyconfigs/test_ecs/t/toy/toy-0.0-test.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    'toy-0.0_fix-silly-typo-in-printf-statement.patch',
+    ('toy-extra.txt', 'toy-0.0'),
+]
+
+sanity_check_paths = {
+    'files': [('bin/yot', 'bin/toy')],
+    'dirs': ['bin'],
+}
+
+runtest = "make_test dummy_cmd" # Provide some value which is unique enough to be checked for
```
Comment (ocaisa) [inline test/framework/easyconfigs/test_ecs/t/toy/toy-0.0-test.eb:30]: Updating the test suite for this file is going to be a bit painful. Maybe just add this line to an existing test eb and use that? Of course this could also have negative consequences...

Comment (ocaisa) [reply test/framework/easyconfigs/test_ecs/t/toy/toy-0.0-test.eb:30]: Maybe not, I see you only had two failures

Comment (Flamefire) [reply test/framework/easyconfigs/test_ecs/t/toy/toy-0.0-test.eb:30]: What do you mean by updating the test suite? What else (besides this PR) needs to be done?

I haven't found any other test_ec with a `runtest` key and adding to an existing might be problematic as some of them are actually installed so they would choke on the "faulty" command.

Comment (ocaisa) [reply test/framework/easyconfigs/test_ecs/t/toy/toy-0.0-test.eb:30]: One of your failing tests is
```
FAIL: test_index_functions (test.framework.filetools.FileToolsTest)
```
which does some kind of counting test. When you add new files to the test suite, these kinds of tests need to be updated (even though they are not related). Your case may not be so bad, but I remember this being tedious for me in the past.

Comment (ocaisa) [reply test/framework/easyconfigs/test_ecs/t/toy/toy-0.0-test.eb:30]: The other failing test is a required update to https://github.com/easybuilders/easybuild-framework/blob/develop/test/framework/docs.py#L248

Comment (Flamefire) [reply test/framework/easyconfigs/test_ecs/t/toy/toy-0.0-test.eb:30]: I see, thanks!

**Review by ocaisa — APPROVED** [review]

**Comment by boegel** [comment]
I'll squeeze this in for EasyBuild v4.3.2...
